rust/worker/src/execution/operator.rs

```diff
@@ -17,16 +22,45 @@ where
     async fn run(&self, input: &I) -> Result<O, Self::Error>;
 }
 
+#[derive(Debug, Error)]
+pub(super) enum TaskError<Err> {
```

Comment:
may want to constrain Err to ChromaError here

Reply:
adding that bound results in needing to add a lot of spurious typings, so imo it's ok to leave it unbounded and introduce it if there's an issue later
anywhere that calls `.boxed()` or `.code()` already effectively constrains it